nengo/builder/processes.py

```diff
@@ -176,3 +181,54 @@ def build_process(model, process, sig_in=None, sig_out=None, mode="set"):
     )
 
     return sig_out
+
+
+@Builder.register(Synapse)
+def build_synapse(
+    model, synapse, sig_in=None, sig_out=None, mode="set", y0=0, **kwargs
+):
+    """Builds a `.Synapse` object into a model.
+
+    Wrapper around `.build_process` that accepts the initial value ``y0``.
+
+    Parameters
+    ----------
+    model : Model
+        The model to build into.
+    process : Process
+        Process to build.
+    sig_in : Signal, optional
+        The input signal, or None if no input signal.
+    sig_out : Signal, optional
+        The output signal, or None if no output signal.
+    mode : "set" or "inc" or "update", optional
+        The ``mode`` of the built `.SimProcess`.
+    y0 : array_like
+        Initial value(s) for the synapse.
+
+    Notes
+    -----
+    Does not modify ``model.params[]`` and can therefore be called
+    more than once with the same `.Synapse` instance.
+    """
+    if sig_out is None:
+        assert sig_in is not None, "Both `sig_in` and `sig_out` cannot be None"
+        output_shape = sig_in.shape
+
+        y0 = np.asarray(y0, dtype=rc.float_dtype)
+        if y0.ndim > 0:
+            assert y0.shape == output_shape
+        else:
+            y0 = y0 * np.ones(output_shape, dtype=rc.float_dtype)
```

Comment:
Why ones?

Reply:
If `y0` is a scalar, broadcast it across all dimensions.

Reply:
Ah right. Thanks